### PR TITLE
Added AE, PP amount to creating new artwork and burn edition method

### PR DESCRIPTION
**Pull request by tienngovan**
Story links:
[Add burnable and bridgeable in the FF exhibition ETH contract#1100](https://github.com/bitmark-inc/feral-file/issues/1100)
[Edition Structure change for ETH contract#1132](https://github.com/bitmark-inc/feral-file/issues/1132)

### CODE REVIEW

File: contracts/FeralfileArtworkV3.sol [hunk 1]
```diff
@@ -394,9 +419,12 @@ contract FeralfileExhibitionV3 is ERC721Enumerable, Authorizable, IERC2981 {
             artworks[artworkID_].editionSize > 0,
             "FeralfileExhibitionV3: artwork is not found"
         );
-        /// @notice The range of editionNumber should be between 0 (AP) ~ artwork.editionSize
+        /// @notice The range of editionNumber should be between 0 to artwork.editionSize + artwork.aeAmount + artwork.ppAmount
         require(
-            editionNumber_ <= artworks[artworkID_].editionSize,
+            editionNumber_ <=
```
Comment (lemonlatte) [inline contracts/FeralfileArtworkV3.sol]: I think it should be 
```suggestion
            editionNumber_ <
```

File: contracts/FeralfileArtworkV3.sol [hunk 2]
```diff
@@ -425,10 +453,36 @@ contract FeralfileExhibitionV3 is ERC721Enumerable, Authorizable, IERC2981 {
...
+        for (uint256 i = 0; i < editionIDs_.length; i++) {
+            require(
+                _exists(editionIDs_[i]),
+                "ERC721: artwork edition is not found"
+            );
+            require(
+                _isApprovedOrOwner(_msgSender(), editionIDs_[i]),
+                "ERC721: caller is not token owner nor approved"
+            );
+            ArtworkEdition storage edition = artworkEditions[editionIDs_[i]];
```
Comment (lemonlatte) [inline contracts/FeralfileArtworkV3.sol]: I think 
```
ArtworkEdition memory edition
```
might save a little bit of cost

File: contracts/FeralfileArtworkV3.sol [hunk 3]
```diff
@@ -425,10 +453,36 @@ contract FeralfileExhibitionV3 is ERC721Enumerable, Authorizable, IERC2981 {
...
+        }
+    }
+
     event NewArtwork(uint256 indexed artworkID);
     event NewArtworkEdition(
         address indexed owner,
         uint256 indexed artworkID,
         uint256 indexed editionID
     );
+    event BurnedEdition(uint256 indexed editionID);
```
Comment (lemonlatte) [inline contracts/FeralfileArtworkV3.sol]:
```
BurnArtworkEdition
```
Suggest to use this event name.

File: contracts/FeralfileArtworkV3.sol [hunk 4]
```diff
@@ -394,9 +419,12 @@ contract FeralfileExhibitionV3 is ERC721Enumerable, Authorizable, IERC2981 {
             artworks[artworkID_].editionSize > 0,
             "FeralfileExhibitionV3: artwork is not found"
         );
-        /// @notice The range of editionNumber should be between 0 (AP) ~ artwork.editionSize
+        /// @notice The range of editionNumber should be between 0 to artwork.editionSize + artwork.aeAmount + artwork.ppAmount
```
Comment (lemonlatte) [inline contracts/FeralfileArtworkV3.sol]:
```suggestion
        /// @notice The range of editionNumber should be between 0 to artwork.editionSize + artwork.aeAmount + artwork.ppAmount - 1
```

**Review by lemonlatte — CHANGES_REQUESTED** [review]

File: contracts/FeralfileArtworkV3.sol (hunk 1 above)
Comment (tienngovan) [reply contracts/FeralfileArtworkV3.sol]: Fixed

File: contracts/FeralfileArtworkV3.sol (hunk 2 above)
Comment (tienngovan) [reply contracts/FeralfileArtworkV3.sol]: Fixed

File: contracts/FeralfileArtworkV3.sol (hunk 3 above)
Comment (tienngovan) [reply contracts/FeralfileArtworkV3.sol]: Fixed

File: contracts/FeralfileArtworkV3.sol (hunk 4 above)
Comment (tienngovan) [reply contracts/FeralfileArtworkV3.sol]: Fixed

File: contracts/FeralfileArtworkV3.sol [hunk 5]
```diff
@@ -107,12 +124,16 @@ contract FeralfileExhibitionV3 is ERC721Enumerable, Authorizable, IERC2981 {
...
+        uint256 editionSize,
+        uint256 aeAmount,
+        uint256 ppAmount
     ) private {
         require(bytes(title).length != 0, "title can not be empty");
         require(bytes(artistName).length != 0, "artist can not be empty");
         require(bytes(fingerprint).length != 0, "fingerprint can not be empty");
         require(editionSize > 0, "edition size needs to be at least 1");
+        require(aeAmount > 0, "AE amount needs to be at least 1");
+        require(ppAmount > 0, "PP amount needs to be at least 1");
```
Comment (lemonlatte) [inline contracts/FeralfileArtworkV3.sol]: Both `aeAmount` and `ppAmount` can be zero. We can remove this check

**Review by lemonlatte — CHANGES_REQUESTED** [review]

File: contracts/FeralfileArtworkV3.sol (hunk 5 above)
Comment (tienngovan) [reply contracts/FeralfileArtworkV3.sol]: Fixed

**Review by lemonlatte — APPROVED** [review]
LGTM

**Pull request merged**